src/kudu/consensus/consensus_meta.cc

```diff
@@ -178,6 +188,12 @@ const LastKnownLeader& ConsensusMetadata::last_known_leader() const {
   return last_known_leader_;
 }
 
+const std::map<int64_t, PreviousVotePB>&
+ConsensusMetadata::previous_vote_history() const {
+  DFAKE_SCOPED_RECURSIVE_LOCK(fake_lock_);
+  return previous_vote_history_;
```

Comment:
Same question: is it safe to return a reference to `previous_vote_history_`

Reply:
I will update it to a copy as well since you rightly pointed out that returning a reference here defeats the purpose.